### DISCUSSION TITLE
Wrong number of columns reported by `tput cols` early on with new window

### DISCUSSION THREAD

**Discussion by ZoltanLajosKis** [ghostty-org/ghostty#3699] [Feature Requests, Ideas] [answered]
### Issue

When opening a new window, `tput cols` returns unexpected values in the first 20 milliseconds (YMMV).

Note: originally noticed using Statship prompt, where the `$fill` module occasionally only filled the initial prompt to 80 characters instead of the expected full length of 120 characters.

### Config to reproduce

`~/.config/ghostty/config`:
```
window-width = 120
window-height = 45
```

(Tested both with and without shell integration.)

`~/.bashrc`:
```
tput cols
tput cols
... ...
tput cols
tput cols
```

(Alternatively use `sleep` with different timeout values and a simple `tput cols` command.)

### Output

Typical output is:
```
80 [repeated 5-6 times]
99 [repeated 18-20 times]
120 [repeated]
```

Occasionally 99 is missing from the output (only 80 and 120 is printed) and rarely 99 is replaced by 79 in the output.

### Versions

Ubuntu 22.04.5 and 24.04.1
Ghostty 1.0.0, built with zig-0.1.3 (-release=fast)

**Comment by mitchellh** [comment, marked as answer]
It's not incorrect. We initialize the terminal before the GUI is initialized. The terminal state does support the size given. Then we'll issue a resize event (SIGWNCH and others) shortly after with the correct size. This lets applications initialize faster since most don't depend on size.

**Reply by ZoltanLajosKis** [reply]
Understood, thank you for the explanation.

Apparently, when Starship calculates the initial prompt size, it reads the column size at different stages of this process. Here is how my terminals look like with Ghostty:

![image](https://github.com/user-attachments/assets/76a3671d-63aa-4a4a-8ef1-24dd709996cb)

**Comment by ZoltanLajosKis** [comment]
I no longer observe this behavior on Ubuntu 24.04 with Ghostty 1.1.3.

**Reply by 00-kat** [reply]
This bug was fixed a little while ago, I don't remember if it was cherry-picked into 1.1.3 though.  That's probably why Ghostty exhibits that behaviour.
